components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager;
+import org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor;
+import org.apache.http.nio.reactor.ConnectingIOReactor;
+import org.apache.http.ssl.SSLContexts;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants;
+import org.wso2.identity.event.websubhub.publisher.exception.WebSubAdapterException;
+import org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil;
+
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import javax.net.ssl.SSLContext;
+
+import static org.apache.http.HttpHeaders.ACCEPT;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ErrorMessages.ERROR_PUBLISHING_EVENT_INVALID_PAYLOAD;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.Http.CORRELATION_ID_REQUEST_HEADER;
+
+/**
+ * Class to retrieve the HTTP Clients.
+ */
+public class ClientManager {
+
+    private static final Log LOG = LogFactory.getLog(ClientManager.class);
+    private final CloseableHttpAsyncClient httpAsyncClient;
+
+    /**
+     * Creates a client manager.
+     *
+     * @throws WebSubAdapterException on errors while creating the http client.
+     */
+    public ClientManager() throws WebSubAdapterException {
+
+        PoolingNHttpClientConnectionManager connectionManager;
+        try {
+            connectionManager = createPoolingConnectionManager();
+            LOG.debug("Successfully created PoolingNHttpClientConnectionManager");
+        } catch (IOException e) {
+            throw WebSubHubAdapterUtil.handleServerException
+                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_ASYNC_HTTP_CLIENT, e);
+        }
+
+        RequestConfig config = createRequestConfig();
+        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+        addSslContext(httpClientBuilder);
+        httpClientBuilder.setConnectionManager(connectionManager);
+        httpAsyncClient = httpClientBuilder.build();
+        httpAsyncClient.start();
+        LOG.debug("HttpAsyncClient started");
+    }
+
+    /**
+     * Get HTTP client properly configured with tenant configurations.
+     *
+     * @return CloseableHttpAsyncClient instance.
+     */
+    public CloseableHttpAsyncClient getClient() {
+
+        if (!httpAsyncClient.isRunning()) {
+            LOG.warn("HttpAsyncClient is not running, starting client");
```

Comment:
IMO, we don't need a warn log here. We can keep this a debug as well

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/7999427bc5211016a93107fb27f9efd67513d525